### PR TITLE
Added the APIKeys interface

### PR DESCRIPTION
**Pull request by StoyanTc**
Introduced the `APIKeys` interface and the `validate()` required by [CEP-158](https://macrometa.atlassian.net/browse/CEP-158) for api keys support in `c8cep`.

### CODE REVIEW

File: src/main/java/com/c8db/C8ApiKeys.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2021 Macrometa Corp All rights reserved.
+ */
+
+package com.c8db;
+
+import com.c8db.entity.ApiKeyEntity;
+import com.c8db.entity.FeaturesEntity;
```
Comment (yehan-macrometa) [inline src/main/java/com/c8db/C8ApiKeys.java:8]: A few unused imports from this line onwards.

File: src/main/java/com/c8db/internal/C8AdminImpl.java [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 import com.c8db.C8Admin;
 import com.c8db.C8DBException;
+import com.c8db.entity.ApiKeyEntity;
```
Comment (yehan-macrometa) [inline src/main/java/com/c8db/internal/C8AdminImpl.java:11]: Unused import.
We can keep this file unchanged in this PR if we remove the blank-line changes as well.

File: src/main/java/com/c8db/internal/InternalC8Admin.java [hunk 3]
```diff
@@ -9,11 +9,17 @@
 import com.arangodb.velocypack.Type;
 import com.arangodb.velocypack.VPackSlice;
 import com.arangodb.velocypack.exception.VPackException;
+import com.c8db.entity.ApiKeyEntity;
```
Comment (yehan-macrometa) [inline src/main/java/com/c8db/internal/InternalC8Admin.java:12]: Unused imports. A few others are below also.
We can keep this file unchanged in this PR if we remove the blank-line changes as well.

**Review by yehan-macrometa — APPROVED** [review]

File: pom.xml [hunk 4]
```diff
@@ -1,329 +1,329 @@
...
+				<version>${hamcrest-all.version}</version>
+			</dependency>
+		</dependencies>
+	</dependencyManagement>
+
+	<scm>
+		<url>https://github.com/Macrometacorp/C84j.git</url>
+		<connection>scm:git:git@github.com:Macrometacorp/C84j.git</connection>
+		<developerConnection>scm:git:git@github.com:Macrometacorp/C84j.git</developerConnection>
+		<tag>c84j-1.1.0-SNAPSHOT</tag>
```
Comment (grainier-macrometa) [inline pom.xml:321]: Shall we update this scm tag to the latest version tag as well?

File: src/main/java/com/c8db/C8Database.java [hunk 5]
```diff
@@ -669,4 +669,10 @@ GraphEntity createGraph(String name, Collection<EdgeDefinition> edgeDefinitions,
      */
     C8Admin admin();
 
+    /**
+     * Returns a {@code C8ApiKeys} instance.
+     *
+     * @return C8Admin handler
```
Comment (grainier-macrometa) [inline src/main/java/com/c8db/C8Database.java:675]: should be `@return C8ApiKeys instance`

File: src/main/java/com/c8db/internal/C8ApiKeyImpl.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import com.c8db.entity.ApiKeyEntity;
+
+public class C8ApiKeyImpl extends InternalC8ApiKeys<C8DBImpl, C8DatabaseImpl, C8ExecutorSync>
+        implements C8ApiKeys {
+
+    protected C8ApiKeyImpl(final C8DatabaseImpl db) {
+        super(db);
+    }
+
+	@Override
```
Comment (grainier-macrometa) [inline src/main/java/com/c8db/internal/C8ApiKeyImpl.java:17]: Small formatting/indentation issue

File: src/main/java/com/c8db/entity/ApiKeyEntity.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+package com.c8db.entity;
+
+import lombok.Data;
+
+@Data
+public class ApiKeyEntity {
+    private String keyid;
+    private String tenant;
+    private String parent;
+    private int limitRequestsPerSec;
```
Comment (grainier-macrometa) [inline src/main/java/com/c8db/entity/ApiKeyEntity.java:13]: @StoyanTc Can we use this information to get the user name?

File: src/main/java/com/c8db/C8ApiKeys.java (hunk 1 above)
Comment (StoyanTc) [reply src/main/java/com/c8db/C8ApiKeys.java:8]: Fixed

File: src/main/java/com/c8db/internal/C8AdminImpl.java (hunk 2 above)
Comment (StoyanTc) [reply src/main/java/com/c8db/internal/C8AdminImpl.java:11]: Fixed

File: pom.xml (hunk 4 above)
Comment (StoyanTc) [reply pom.xml:321]: Done

File: src/main/java/com/c8db/C8Database.java (hunk 5 above)
Comment (StoyanTc) [reply src/main/java/com/c8db/C8Database.java:675]: Fixed

File: src/main/java/com/c8db/entity/ApiKeyEntity.java (hunk 7 above)
Comment (StoyanTc) [reply src/main/java/com/c8db/entity/ApiKeyEntity.java:13]: @grainier-macrometa - Yes. The parent is actually the `tenant` + '." + `user`.

**Review by chandrashekhar-macrometa — APPROVED** [review]

**Review by grainier-macrometa — APPROVED** [review]

**Pull request merged**